src/libs/Navigation/AppNavigator/AuthScreens.tsx

```diff
@@ -109,6 +107,21 @@ Onyx.connect({
     },
 });
 
+Onyx.connect({
+    key: ONYXKEYS.ONYX_UPDATES_LAST_UPDATE_ID_APPLIED_TO_CLIENT,
+    callback: (value: OnyxEntry<number>) => {
+        lastUpdateIDAppliedToClient = value;
```

Comment:
If @situchan 's [finding](https://github.com/Expensify/App/pull/32776#issuecomment-1876896706) is really a problem, maybe you can resolve a promise here so you can hold on calling App.reconnectApp until the value is set.... but then I wonder if there may be cases where we never set `ONYX_UPDATES_LAST_UPDATE_ID_APPLIED_TO_CLIENT` (accounts with reliable updates disabled?)